src/dc_receive_imf.rs

```diff
@@ -5046,4 +5049,101 @@ Reply from different address
 
         Ok(())
     }
+
+    #[async_std::test]
+    async fn test_outgoing_private_reply_multidevice() -> Result<()> {
+        let mut tcm = TestContextManager::new().await;
+        let alice1 = tcm.alice().await;
+        let alice2 = tcm.alice().await;
+        let bob = tcm.bob().await;
+
+        // =============== Bob creates a group ===============
+        let group_id =
+            chat::create_group_chat(&bob, ProtectionStatus::Unprotected, "Group").await?;
+        chat::add_to_chat_contacts_table(
+            &bob,
+            group_id,
+            bob.add_or_lookup_contact(&alice1).await.id,
+        )
+        .await?;
+        chat::add_to_chat_contacts_table(
+            &bob,
+            group_id,
+            Contact::create(&bob, "", "charlie@example.org").await?,
+        )
+        .await?;
+
+        // =============== Bob sends the first message to the group ===============
+        let sent = bob.send_text(group_id, "Hello all!").await;
+        alice1.recv_msg(&sent).await;
+        alice2.recv_msg(&sent).await;
+
+        // =============== Alice answers privately with device 1 ===============
+        let received = alice1.get_last_msg().await;
+        let alice1_bob_contact = alice1.add_or_lookup_contact(&bob).await;
+        assert_eq!(received.from_id, alice1_bob_contact.id);
+        assert_eq!(received.to_id, DC_CONTACT_ID_SELF);
+        assert_eq!(received.viewtype, Viewtype::Text);
+        assert!(!received.state.is_outgoing());
+        assert!(!received.hidden);
+        assert_eq!(received.text, Some("Hello all!".to_string()));
+        assert_eq!(received.in_reply_to, None);
+        assert_eq!(received.is_dc_message, MessengerMessage::Yes);
+        assert_eq!(received.mime_modified, false);
+        assert_eq!(received.chat_blocked, Blocked::Request);
+        assert_eq!(received.error, None);
+        assert_eq!(received.param.get(Param::OverrideSenderDisplayname), None);
+        assert_eq!(received.param.get(Param::File), None);
+        assert_eq!(received.param.get(Param::Bot), None);
```

Comment:
i'm pretty sure you don't care about **all** these asserts here.  maybe trim them down to just the few that mater for this?
(same for other blocks of asserts below)

Reply:
Like this?

Reply:
:+1: